workflow/table_generation.py

```diff
@@ -0,0 +1,800 @@
+            if pep2Prize[parts[0]] >= sigThresh:
+                prot2TimeSeries["sig"][prot].append(timeSeries)
+            else:
+                prot2TimeSeries["insig"][prot].append(timeSeries)
+
+    print("Loaded time series for %d proteins (and pseudonodes)" %
+          len(prot2TimeSeries["all"]))
+    # Only some datasets need an initial 0 value to be added, some may already
+    # include it
+    if addZero:
+        print("Added 0 to the start of every time series")
+    print("Min value in time series: %f" % timeSeriesMin)
+    print("Max value in time series: %f" % timeSeriesMax)
+
+    # Statistics on the peptide to protein mappings for all, significant,
+    # and significant peptides
+    pepsPerProt = dict()
+    for timeSeriesType in prot2TimeSeries.keys():
+        pepsPerProt[timeSeriesType] = map(
+            len, prot2TimeSeries[timeSeriesType].values())
+#     print("Mean peptides per protein: %f" % np.mean(pepsPerProt["all"]))
+#     print("Max peptides per protein: %d" % max(pepsPerProt["all"]))
+#     print("Max significant peptides per protein: %d" % max(pepsPerProt["sig"]))
+#     print("Max insignificant peptides per protein: %d" %
+#           max(pepsPerProt["insig"]))
+
+    # Load and summarize the temporal activity windows
+    # The map from proteins to all windows maps to a list where there is a set
+    # of inferred activity states for each time point/window.  These are
+    # collapsed later
+    prot2AllWindows = dict()
+    with open(windowsFile) as f:
+        # Use the header to get the time points of the activity windows
+        header = next(f).strip().split("\t")
+        numCols = len(header)
+        times = header[1:]
+        for line in f:
+            parts = line.strip().split("\t")
+            if len(parts) != numCols:
+                raise RuntimeError(
+                    "All activity window lines must have %d columns\n%s" % (numCols, line))
+            # The protein id may be followed by the peptide id
+            prot = parts[0].split("#")[0]
+            windows = prot2AllWindows.setdefault(
+                prot, [set() for i in range(len(times))])
+            for i, state in enumerate(parts[1:]):
+                windows[i].add(state)
+
+    print("Loaded activity windows for %d proteins" % len(prot2AllWindows))
+
+    # Summarize the temporal activity windows and convert them to a numerical
+    # format.  Also store the time point when the protein is first active
+    # (activation/inhibition/ambiguous state)
+    prot2WindowsSummary = dict()
+    prot2FirstActive = dict()
+    for prot in prot2AllWindows.keys():
+        prot2WindowsSummary[prot] = map(ConvertWindowState, map(
+            SummarizeWindow, prot2AllWindows[prot]))
+        prot2FirstActive[prot] = FirstActiveGeneral(
+            prot2WindowsSummary[prot], times)
+
+    # Load the list of gold standard proteins
+    goldStandard = set()
+    with open(goldStandardFile) as f:
```

Comment:
Here the code tries to open `goldStandardFile`. If we wrap this in a conditional, we can simply skip over the file opening and for loop if the file was not provided.  It could default to `None`.  Then the `goldStandard` will be the empty set, and the rest of the code should work.